cylc/flow/platforms.py

```diff
@@ -74,7 +74,10 @@ def get_platform(task_conf=None, task_id='unknown task', warn_only=False):
             # In warning mode this function might have been passed an
             # un-expanded platform string - warn that they won't deal with
             # with this until job submit.
-            return None
+            return (
+                f"{task_conf['platform']} will be evaluated at Job Submission "
+                "and may fail then."
+            )
```

Comment:
I think this was removed after a review comment.

This would raise a warning every time someone tries to use a subshell in a platforms item telling the user that the command could fail?

If so this is a bit over the top, a warning in the docs should be more than sufficient.